File: Python/Objects/item.py

class Item: 
    def __init__(self,UPC,price,name,):
        self.UPC = UPC
        self.price = price
        self.name = name

    def setUPC(self,UPC):
        if UPC is int and len(UPC==12):
            self.UPC = UPC
        else:
            raise Exception("INVALID UPC")

    def getUPC(self):
        return self.UPC

    def setPrice(self,price):
        if price is float:
            self.price = round(price,2)
        else:
            raise Exception ("INVALID PRICE")

    def getPrice(self):
        return self.price

    def setName(self,name):
        #Needs more checks
        if name is str:
            self.name = name
        else:
            raise Exception("INVALID NAME")

    def getName(self):
        return self.name

File: Dart/web_scraper_app/lib/objects/item.dart
class Item {
  // ignore: non_constant_identifier_names
  int UPC;
  double price;
  String name;
  String store;

  Item(this.UPC, this.price, this.name, this.store);

  // ignore: non_constant_identifier_names
  bool setUPC(int UPC) {
    if (UPC is int) {
      this.UPC = UPC;
      return true;
    } else {
      return false;
    }
  }

  int getUPC() {
    return this.UPC;
  }

  bool setPrice(double price) {
    if (price is double) {
      this.price = price;
      return true;
    } else {
      return false;
    }
  }

  double getPrice() {
    return this.price;
  }

  bool setName(String name) {
    if (name is String) {
      this.name = name;
      return true;
    } else {
      return false;
    }
  }

  String getName() {
    return this.name;
  }

  bool setStore(String store) {
    if (store is String) {
      this.store = store;
      return true;
    } else {
      return false;
    }
  }

  String getStore() {
    return this.store;
  }

  factory Item.fromJson(Map<String, dynamic> processedJson) {
    return Item(processedJson["UPC"], processedJson["price"],
        processedJson["name"], processedJson["store"]);
  }
}
